ca/certificate-authority.go

```diff
@@ -277,18 +281,31 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 	commonName := ""
 	hostNames := make([]string, len(csr.DNSNames))
 	copy(hostNames, csr.DNSNames)
+	if len(csr.Subject.CommonName) > maxCNLength {
+		msg := fmt.Sprintf("Common name was longer than 64 bytes, was %d",
+			len(csr.Subject.CommonName))
+		err := core.MalformedRequestError(msg)
+		// AUDIT[ Certificate Requests ] 11917fa4-10ef-4e0d-9105-bacbe7836a3c
+		ca.log.AuditErr(err)
+		return emptyCert, err
+	}
+
 	if len(csr.Subject.CommonName) > 0 {
 		commonName = strings.ToLower(csr.Subject.CommonName)
 		hostNames = append(hostNames, commonName)
-	} else if len(hostNames) > 0 {
-		commonName = strings.ToLower(hostNames[0])
-	} else {
+	}
+
+	if len(hostNames) == 0 {
 		err = core.MalformedRequestError("Cannot issue a certificate without a hostname.")
 		// AUDIT[ Certificate Requests ] 11917fa4-10ef-4e0d-9105-bacbe7836a3c
 		ca.log.AuditErr(err)
 		return emptyCert, err
 	}
 
+	if ca.forceCNFromSAN && commonName == "" {
```

Comment:
I don't want to cycle through and pick a name, I agree that's too much complexity. However, I do want this fix to make sure we're compliant in all cases. Let's move the `if len(csr.Subject.CommonName) > maxCNLength` check to below here, so that we can reject in any case where we would have issued with a CN > 64 bytes.